### PR TITLE
JAPAN-289 Added character restriction validation

### PR DESCRIPTION
**Pull request by toddlawton**

### CODE REVIEW

File: html5-validation-engine.js [hunk 1]
```diff
@@ -296,6 +305,17 @@
                     isNotValid = false,
                     type ="";
 
+                if (characterRestrictionPattern){
+                    var regex = new RegExp(characterRestrictionPattern);
+                    isNotValid = isNotValid || (!regex.test($input.val()) ? true : false);
```
Comment (nebez) [inline html5-validation-engine.js]: [RegExp.test()](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test) returns a boolean already. The ternary is a little redundant in this case (!bool) ? true : false;

File: html5-validation-engine.js [hunk 2]
```diff
@@ -296,6 +305,17 @@
                     isNotValid = false,
                     type ="";
 
+                if (characterRestrictionPattern){
```
Comment (nebez) [inline html5-validation-engine.js]: forgot a space between ) and {

**Comment by toddlawton** [comment]
Fixed!

**Comment by nebez** [comment]
reviewed.

**Pull request merged**